File: Neetcode/LongestSubStrNoRepeat.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l = 0
        r = 0
        wordset = set()
        maxsize = 0
        while r < len(s):
            if wordset:
                if s[r] in wordset:
                    while s[l] != s[r]:
                        wordset.remove(s[l])
                        l += 1
                    l += 1
                else:
                    wordset.add(s[r])
            else:
                wordset.add(s[r])
            maxsize = max(maxsize, r-l + 1)
            r += 1
        return maxsize

File: Neetcode/LongestSubStrNoRepeat.cpp
#include <unordered_set>
using std::unordered_set;
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        cout << s.length();
        int left = 0;
        int maxlen = 0;
        int i=0;
        unordered_set<char> hashset;
        for (; i<s.length(); i++){
            if (hashset.find(s[i]) != hashset.end()){
                maxlen = max(maxlen, i-left);
                while (s[left] != s[i]){
                    hashset.erase(s[left]);
                    left += 1;
                }
                left += 1;
            }
            else{
                hashset.insert(s[i]);
            }
        }
        cout << left << i;
        maxlen = max(maxlen, i-left);
        return maxlen;
    }
};
